action/protocol/staking/protocol.go

```diff
@@ -169,26 +166,35 @@ func (p *Protocol) CreateGenesisStates(
 			return err
 		}
 	}
-	return nil
+
+	// commit updated view
+	if err := csm.Commit(); err != nil {
+		return errors.Wrap(err, "failed to commit candidate change")
+	}
+
+	// write update view back to state factory
+	return csm.WriteView(protocolID, csm.CandCenter())
 }
 
 // Commit commits the last change
 func (p *Protocol) Commit(ctx context.Context, sm protocol.StateManager) error {
-	center, err := p.getCandCenter(sm.ConfirmedHeight())
+	csm, err := NewCandidateStateManager(sm)
 	if err != nil {
-		return errors.Wrap(err, "failed to commit candidate change in sm, cand center does not exist")
+		return err
 	}
 
-	csm, err := NewCandidateStateManager(sm, center)
-	if err != nil {
-		return errors.Wrap(err, "failed to commit candidate change in sm")
+	// commit updated view
+	if err := csm.Commit(); err != nil {
+		return errors.Wrap(err, "failed to commit candidate change")
 	}
-	return csm.Commit()
+
+	// write update view back to state factory
+	return csm.WriteView(protocolID, csm.CandCenter())
```

Comment:
WriteView should be called in csm.Commit

Reply:
yeah, can do